src/BenchmarkDotNet/Code/DeclarationsProvider.cs

```diff
@@ -62,6 +62,21 @@ public IEnumerable<string> ArgumentsNamespaces
             => Descriptor.WorkloadMethod.GetParameters()
                 .Where(arg => !string.IsNullOrEmpty(arg.ParameterType.Namespace))
                 .Select(arg => $"using {arg.ParameterType.Namespace};");
+
+        private string GetMethodName(MethodInfo method)
+        {
+            if (method == null)
+            {
+                return EmptyAction;
+            }
+
+            if (method.ReturnType != typeof(Task))
```

Comment:
an asynchronous Setup method should be returning a `Task`. However knowing our users some of them will also return a `Task<T>` or even a `ValueTask<T>`.

Could you please handle these two cases as well?

```cs
if (method.ReturnType.GetTypeInfo().IsGenericType
    && (method.ReturnType.GetTypeInfo().GetGenericTypeDefinition() == typeof(Task<>)
        || method.ReturnType.GetTypeInfo().GetGenericTypeDefinition() == typeof(ValueTask<>)))
```

Reply:
Also added support for Task<T>, ValueTask and ValueTask<T>.